### PR TITLE
chore: miscellaneous log4j tasks + zcs-lib and ldap-utilities moved in here

### PR DESCRIPTION
**Pull request by frisonisland**
**Moved repos content:**
- https://github.com/Zextras/carbonio-zcs-lib moved here. Reason is this is a library consisting of dependencies only, directly tied to mailbox and must be always aligned. Also helps avoiding issue in building/testing.
Library name is **carbonio-jetty-libs** and **store-extra-runtime-dependencies was removed since it was a copy of it basically**.

- Partially moved https://github.com/Zextras/carbonio-ldap-utilities here. Moved LDAP template files and migrations. Only utils tools are left in the other repo. **See content in: store/ldap**.
**LDAP schema is generated in prepare-package in ant-store.xml**.
See carbonio-ldap-utilities PR: https://github.com/zextras/carbonio-ldap-utilities/pull/52

See this related PRs for the build process affected by migration: 
- https://github.com/zextras/carbonio-build/compare/devel...refactor/ldap-utilities-mailbox

**Log4J changes:**
- **log4j.properties.production**: if zimbraLogHostname is not set, slogger configuration fails with ERROR message because host value cannot be empty
- zmmailboxdctl uses -Dlog4j.configurationFile: https://github.com/zextras/carbonio-core-utils/pull/52/files

**Other:**
- changed default UserAgent in SoapTransport to "Carbonio"

### CODE REVIEW

File: carbonio-jetty-libs/pom.xml [hunk 1]
```diff
@@ -0,0 +1,816 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns="http://maven.apache.org/POM/4.0.0"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <artifactId>zm-mailbox</artifactId>
+    <groupId>zextras</groupId>
```
[the remark this answers is not in the document]
Comment (frisonisland) [reply carbonio-jetty-libs/pom.xml:8]: It should be probably, when I made ant -> maven conversion I used zextras, so all other poms have group zextras. I was not aware of the standard group we are using

**Comment by keshavbhatt** [comment]
Wouldn't it have been nice to keep these PRs separate? Is there any specific reason why we need to mix the repo move with the log4j 2 migration?

**Comment by frisonisland** [comment]
> Wouldn't it have been nice to keep these PRs separate? Is there any specific reason why we need to mix the repo move with the log4j 2 migration?

Because the whole process was a chore :cry: repo status was preventing me from building a consistent state of the system with log4j, thus preventing me from testing it

**Review by keshavbhatt — APPROVED** [review]

File: carbonio-jetty-libs/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Polpetta) [reply carbonio-jetty-libs/pom.xml:8]: Imo it should be `com.zextras.mailbox` for everything mailbox-related. I used `com.zextras.license` for everything license related

**Review by Polpetta — APPROVED** [review]

**Pull request merged**